03-fetch/readme.md

```diff
@@ -1,292 +1,293 @@
+  user: UserEntity;
+}
+
+export const UserRow = (props: Props) =>
+  <tr>
+    <td>
+      <img src={props.user.avatar_url} style={{ maxWidth: '10rem' }} />
+    </td>
+    <td>
+      <span>{props.user.id}</span>
+    </td>
+    <td>
+      <span>{props.user.login}</span>
+    </td>
+  </tr>
+```
+
+_./pages/components/user-collection/user-table.tsx_
+
+```typescript
+import { UserEntity } from "model/user";
+import { UserHeader } from "./header";
+import { UserRow } from "./row";
+
+interface Props {
+  userCollection: UserEntity[],
+}
+
+export const UserTable = (props : Props) => 
+<table>
+<thead>
+  <UserHeader />
+</thead>
+<tbody>
+  {
+    props.userCollection.map((user: UserEntity) =>
+      <UserRow user={user} key={user.id} />
+    )
+  }
+</tbody>
+</table>
+```
+
+_./pages/components/user-collection/index.ts_
+
+```typescript
+export {UserTable} from './user-table';
+```
+
+- Let's update the page.
+
+_./pages/index.tsx_
+
+```diff
+import * as React from 'react';
+import * as Next from 'next';
+import Link from 'next/link';
+import getUserCollection from '../rest-api/github';
+import { UserEntity } from 'model/user';
++ import {UserTable} from './components/user-collection';
+
+
+interface Props {
+  userCollection: UserEntity[],
+}
+
+const Index : Next.NextSFC<Props> = (props) => (
+  <div>
+  <p>Hello Next.js</p>
+
++  <UserTable userCollection={props.userCollection}/>
+
+  <Link href="/user-info">
+    <a>Navigate to user info page</a>
+  </Link>
+</div>
+)
+
+Index.getInitialProps = async () =>  {
+  const data = await getUserCollection();
+
+  return {
+    userCollection: data,
+  }
+}
+```
+
+## Appendix
+
+What if we want to store the list of users in the component State instead of Properties, we could do it in this way:
+
+```typescript
+import * as React from 'react';
+import Link from 'next/link';
+import getUserCollection from '../rest-api/github';
+import {UserEntity} from '../model/user'
+import { UserHeader } from './components/user-info/header';
+import { UserRow } from './components/user-info/row';
+
+
+interface Props {
+  userCollection: UserEntity[],  
+}
+
+interface State {
+  myStateCollection : UserEntity[];
+}
+
+export class Index extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {myStateCollection: []}
+    console.log("test ****************************");
+  }
+
+  static async getInitialProps() {
+    console.log("getInitialProps ****************************");
+    const data = await getUserCollection();
+  
+    return {
+      userCollection: data,  
+    }
+    
+  }
+
+  static getDerivedStateFromProps(nextProps : Props, prevState : State) : State {    
+    if(nextProps.userCollection !== prevState.myStateCollection) {
+      return ({
+        myStateCollection: nextProps.userCollection,
+      })
+    } else {
+      return null;
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <p>Hello Next.js</p>
+
+        <table>

```

Comment:
Only a suggestion.
At this point we can use _UserTable_ component instead of `<table>....</table>`
```jsx
 <UserTable userCollection={this.state.myStateCollection} />
```

Reply:
Totally agree, please go ahead correct it in the readme, and check that the final version contains this UserTable, thanks for pointing out